qiskit/algorithms/minimum_eigen_solvers/adapt_vqe.py

```diff
@@ -0,0 +1,342 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020, 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""A ground state calculation employing the AdaptVQE algorithm."""
+from __future__ import annotations
+from typing import Optional, List, Tuple, Union
+
+import re
+import logging
+
+from enum import Enum
+import numpy as np
+from qiskit import QiskitError
+
+from qiskit.algorithms import VQE
+from qiskit.algorithms.list_or_dict import ListOrDict
+from qiskit.algorithms.minimum_eigen_solvers.vqe import VQEResult
+from qiskit.algorithms import VariationalAlgorithm
+from qiskit.circuit import QuantumCircuit
+from qiskit.opflow import OperatorBase, PauliSumOp, ExpectationBase, CircuitSampler
+from qiskit.opflow.gradients import GradientBase, Gradient
+from qiskit.circuit.library import EvolvedOperatorAnsatz
+from qiskit.utils.validation import validate_min
+from qiskit.utils import QuantumInstance
+from qiskit.providers import Backend
+from qiskit.algorithms.aux_ops_evaluator import eval_observables
+from ..exceptions import AlgorithmError
+
+
+logger = logging.getLogger(__name__)
+
+
+class Finishing_criterion(Enum):
+    CONVERGED = "Threshold converged"
+    CYCLICITY = "Aborted due to cyclicity of gradient"
+    MAXIMUM = "Maximum number of iterations reached"
+    UNKNOWN = "The algorithm finished due to unforeseen reason"
+
+
+class AdaptVQE(VariationalAlgorithm):
+    """A ground state calculation employing the AdaptVQE algorithm.
```

Comment:
`ground state calculation` is mostly still a referral to Qiskit Nature objects. This should be adapted accordingly.
You could also provide a few more details here, stating that this is an adaptive wrapper of a `VQE` (used internally, not inherited), link to the original paper, etc.

We should also check how this relates to `AdaptQAOA` (https://github.com/Qiskit/qiskit-terra/pull/7131). If I recall correctly, @Cryoris had once suggested that we check, whether this class might even be able to take a QAOA internally.